genesis/engine/entities/rigid_entity/rigid_entity.py

```diff
@@ -1834,6 +1834,32 @@ def set_pos(self, pos, envs_idx=None, *, relative=False, zero_velocity=True, uns
         if zero_velocity:
             self.zero_all_dofs_velocity(envs_idx, unsafe=unsafe)
 
+    @gs.assert_built
+    def get_weld_constraints(self, as_tensor: bool = True, to_torch: bool = True):
+        welds = self.scene.sim.rigid_solver.get_weld_constraints(as_tensor=as_tensor, to_torch=to_torch)
+        obj_a = welds["obj_a"]
+        obj_b = welds["obj_b"]
+        if as_tensor:
+            mask = (obj_a == self.idx) | (obj_b == self.idx)
+            if self._solver.n_envs > 0:
+                welds["valid_mask"] = mask
+            for k in ("obj_a", "obj_b"):
+                welds[k] = welds[k][mask]
+            if "env" in welds:
```

Comment:
Optional keys should be avoided unless strictly necessary. Why do you need this?